### PR TITLE
SONARIAC-802 [Microsoft.Blockchain/*] properties/firewallRules/startI…

### PR DESCRIPTION
**Pull request by mstachniuk**
…pAddress and endIpAddress

### CODE REVIEW

File: iac-extensions/arm/src/test/resources/checks/PublicNetworkAccessCheckTest/rangePublicIPAddress-firewallRules/unknown-type.json
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/arm/src/test/resources/checks/PublicNetworkAccessCheckTest/rangePublicIPAddress-firewallRules/unknown-type.json:1]: I would have limited the check with unknown type to only one case :sweat_smile:

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java [hunk 1]
```diff
@@ -179,12 +183,19 @@ private static boolean isSensitivePublicNetworkAccessSimplified(Tree tree) {
     return TextUtils.matchesValue(tree, "Enabled"::contains).isTrue();
   }
 
-  private static BiConsumer<CheckContext, ResourceDeclaration> checkIpRange() {
-    return (ctx, resource) -> {
-      Optional<Tree> startIpAddress = PropertyUtils.value(resource, "startIpAddress");
-      Optional<Tree> endIpAddress = PropertyUtils.value(resource, "endIpAddress");
-      IpAddressValidator validator = new IpAddressValidator((ArmTree) startIpAddress.orElse(null), (ArmTree) endIpAddress.orElse(null));
-      validator.reportIssueIfPublicIPAddress(ctx, PUBLIC_IP_ADDRESS_MESSAGE, PUBLIC_IP_ADDRESS_MESSAGE_SECONDARY_LOCATION);
+  private static <S extends ContextualMap<S, T>, T extends HasProperties & Tree> void checkIpRange(ContextualMap<S, T> resource) {
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java:186]: I wish there was a way to write this simpler, but I don't think there is.
The only way I found was to write it this way:
`private static void checkIpRange(ContextualMap<?, ?> resource) {`
I'm not convinced it's a good idea, I let you decide on this.

**Review by rudy-regazzoni-sonarsource — APPROVED** [review]
LGTM! Very complete test cases, please see my few comments!

File: iac-extensions/arm/src/test/resources/checks/PublicNetworkAccessCheckTest/rangePublicIPAddress-firewallRules/unknown-type.json
Comment (mstachniuk) [reply iac-extensions/arm/src/test/resources/checks/PublicNetworkAccessCheckTest/rangePublicIPAddress-firewallRules/unknown-type.json:1]: In other cases for other types, there are checks for all cases, so I would keep it.

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java (hunk 1 above)
Comment (mstachniuk) [reply iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java:186]: But then the generic checks are weaker. I will keep it as it is.

**Pull request merged**